botocore/awsrequest.py

```diff
@@ -315,7 +315,7 @@ def create_request_object(request_dict):
     r = request_dict
     request_object = AWSRequest(
         method=r['method'], url=r['url'], data=r['body'], headers=r['headers'])
-    request_object.context.update(r['context'])
+    request_object.context = r['context']
```

Comment:
Why did this change?

Reply:
I added it so that a handler mutating the `context` in the `request-created` event can have the changes be available in subsequent events (e.g. `response-received`). The other option would be to update the `request-created` event to include the `context` but I was trying to avoid expanding the event, especially because the `AWSRequest` object already includes the `context` as a property.